conan/tools/env/environment.py

```diff
@@ -18,37 +18,43 @@ def environment_wrap_command(env_filenames, env_folder, cmd, subsystem=None,
     if not env_filenames:
         return cmd
     filenames = [env_filenames] if not isinstance(env_filenames, list) else env_filenames
-    bats, shs, ps1s = [], [], []
+    bats, shs, ps1s, fishes = [], [], [], []
 
-    accept = accepted_extensions or ("ps1", "bat", "sh")
+    accept = accepted_extensions or ("ps1", "bat", "fish", "sh")
     # TODO: This implemantation is dirty, improve it
     for f in filenames:
         f = f if os.path.isabs(f) else os.path.join(env_folder, f)
         if f.lower().endswith(".sh"):
             if os.path.isfile(f) and "sh" in accept:
                 f = subsystem_path(subsystem, f)
                 shs.append(f)
+        elif f.lower().endswith(".fish"):
```

Comment:
it might be fragile to use extension-based detection.
maybe, it should parse shebang/interpreter as a first try, and only fallback to the extension check if there is no shebang?
that might be worth it, as `.fish` extension is not a standard one like `.sh` and used for few other things (so not every .fish file is a fish script), and vice versa, some shell scripts might have different extension (or might have no extension at all), and it's completely normal in *nix world, where executable files are identified by the interpreter/shebang string instead of extension.